### PR TITLE
[experiment] zero copy BasicStatistics DRAFT

### PR DESCRIPTION
**Pull request by icfaust**
## Description

_Add a comprehensive description of proposed changes_

_List associated issue number(s) if exist(s): #6 (for example)_

_Documentation PR (if needed): #1340 (for example)_

_Benchmarks PR (if needed): https://github.com/IntelPython/scikit-learn_bench/pull/155 (for example)_

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] I have run it locally and tested the changes extensively.
- [ ] All CI jobs are green or I have provided justification why they aren't.
- [ ] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.
- [ ] I have provided justification why quality metrics have changed or why changes are not expected.
- [ ] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

File: onedal/cluster/kmeans.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def _tolerance(self, X_table, rtol, is_csr, dtype):
         bs = self._get_basic_statistics_backend("variance")
 
         res = bs._compute_raw(X_table, dummy, dtype, is_csr)
-        mean_var = from_table(res["variance"]).mean()
+        mean_var = from_table(res.variance).mean()
```
[the remark this answers is not in the document]
Comment (icfaust) [reply onedal/cluster/kmeans.py:108]: I've changed the access pattern.

Comment (ethanglaser) [inline onedal/cluster/kmeans.py:108]: variance_ ?

**Comment by ethanglaser** [comment]
This needs an update: https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/onedal/basic_statistics/tests/utils.py#L35 (or even a refactor)

File: onedal/cluster/kmeans.py (hunk 1 above)
Comment (icfaust) [reply onedal/cluster/kmeans.py:108]: Underscores set in the onedal and sklearnex objects, but the result objects follows oneDAL conventions (and requires the from_table call).

Comment (ethanglaser) [reply onedal/cluster/kmeans.py:108]: Ah, I see, raw_compute vs. compute

**Comment by icfaust** [comment]
> This needs an update: https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/onedal/basic_statistics/tests/utils.py#L35 (or even a refactor)

Good point, in this case the dict is keyed not off the result but off the option, which doesn't have the trailing underscore (if there were problems, there would be test failures all over the place).

**Comment by ethanglaser** [comment]
> > This needs an update: https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/onedal/basic_statistics/tests/utils.py#L35 (or even a refactor)
> 
> Good point, in this case the dict is keyed not off the result but off the option, which doesn't have the trailing underscore (if there were problems, there would be test failures all over the place).

spmd tests in private CI fail as a result of this

**Comment by icfaust** [comment]
> > > This needs an update: https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/onedal/basic_statistics/tests/utils.py#L35 (or even a refactor)
> > 
> > 
> > Good point, in this case the dict is keyed not off the result but off the option, which doesn't have the trailing underscore (if there were problems, there would be test failures all over the place).
> 
> spmd tests in private CI fail as a result of this

You are absolutely right, my bad. Made modifications for that to make it clear to future developers.

**Review by ethanglaser — APPROVED** [review]
